Auto-analysis error during Variadic Function Signature Override step

An error occurred while running auto-analysis on the Sly 1 May 2002 proto. It looked like it happened after the "Variadic Function Signature Override" step began. I believe I had all the analyzer options checked except for "Condense Filler Bytes" and "Elf Scalar Operand References". Here is the stack trace:
```
Data type _reent___new has _reent within it.
java.lang.IllegalArgumentException: Data type _reent___new has _reent within it.
	at ghidra.program.database.data.StructureDB.replaceAtOffset(StructureDB.java:1561)
	at ghidra.emotionengine.symboltable.StdumpAST$InlineStructOrUnion.addField(StdumpAST.java:336)
	at ghidra.emotionengine.symboltable.StdumpAST$InlineStructOrUnion.fill(StdumpAST.java:283)
	at ghidra.emotionengine.symboltable.StabsImporter.importDataTypes(StabsImporter.java:260)
	at ghidra.emotionengine.symboltable.StabsImporter.doImport(StabsImporter.java:177)
	at ghidra.emotionengine.symboltable.StabsAnalyzer.added(StabsAnalyzer.java:84)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:186)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:688)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:788)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:667)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:632)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:58)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:102)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:336)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: ghidra.program.model.data.DataTypeDependencyException: Data type _reent___new has _reent within it.
	at ghidra.program.database.data.CompositeDB.checkAncestry(CompositeDB.java:288)
	at ghidra.program.database.data.StructureDB.replaceAtOffset(StructureDB.java:1485)
	... 15 more

STABS> Type lookup failed: FE
STABS> Type lookup failed: FE
STABS> Type lookup failed: FE
STABS> Type lookup failed: FE
STABS> Type lookup failed: ASO
STABS> Type lookup failed: rtx_def
```

## Additional details:
Plugin version: 2.1.2
Ghidra version: 10.2.3

--- a/src/main/java/ghidra/emotionengine/symboltable/StabsImporter.java
+++ b/src/main/java/ghidra/emotionengine/symboltable/StabsImporter.java
@@ -13,6 +13,7 @@ import java.util.HashSet;
 import java.util.Map;
 
 import ghidra.app.cmd.function.CreateFunctionCmd;
+import ghidra.app.cmd.label.DeleteLabelCmd;
 import ghidra.app.util.exporter.ElfExporter;
 import ghidra.app.util.exporter.ExporterException;
 import ghidra.app.util.importer.MessageLog;
@@ -35,6 +36,7 @@ import ghidra.program.model.listing.Program;
 import ghidra.program.model.listing.Variable;
 import ghidra.program.model.symbol.SourceType;
 import ghidra.program.model.symbol.Symbol;
+import ghidra.program.model.symbol.SymbolIterator;
 import ghidra.program.model.symbol.SymbolTable;
 import ghidra.program.model.util.CodeUnitInsertionException;
 import ghidra.util.exception.DuplicateNameException;
@@ -269,6 +271,8 @@ public class StabsImporter extends FlatProgramAPI {
 		monitor.setMaximum(importer.ast.files.size());
 		monitor.setProgress(0);
 		
+		removeJunkLabels(program);
+		
 		for(int i = 0; i < importer.ast.files.size(); i++) {
 			StdumpAST.SourceFile sourceFile = (StdumpAST.SourceFile) importer.ast.files.get(i);
 			for(StdumpAST.Node functionNode : sourceFile.functions) {
@@ -280,7 +284,8 @@ public class StabsImporter extends FlatProgramAPI {
 					Address high = toAddr(def.addressRange.high - 1);
 					AddressSet range = new AddressSet(low, high);
 					Function function = findOrCreateFunction(def, low, high, range);
-					setFunctionName(function, def, sourceFile, low);
+					setFunctionName(function, def, sourceFile, low, program);
+					function.setComment(sourceFile.path);
 					if(type.returnType != null) {
 						try {
 							function.setReturnType(type.returnType.createType(importer), SourceType.ANALYSIS);
@@ -304,7 +309,23 @@ public class StabsImporter extends FlatProgramAPI {
 			monitor.setProgress(i);
 		}
 	}
-
+	
+	public void removeJunkLabels(Program program) {
+		final String JUNK_LABELS[] = {
+				"__gnu_compiled_c",
+				"__gnu_compiled_cplusplus",
+				"gcc2_compiled."
+		};
+		
+		SymbolTable symbolTable = program.getSymbolTable();
+		for(String junkLabel : JUNK_LABELS) {
+			SymbolIterator iterator = symbolTable.getSymbols(junkLabel);
+			while(iterator.hasNext()) {
+				symbolTable.removeSymbolSpecial(iterator.next());
+			}
+		}
+	}
+	
 	private Function findOrCreateFunction(StdumpAST.FunctionDefinition def, Address low, Address high, AddressSet range) {
 		Function function = getFunctionAt(low);
 		if(function == null) {
@@ -324,27 +345,12 @@ public class StabsImporter extends FlatProgramAPI {
 	}
 	
 	private void setFunctionName(Function function, StdumpAST.FunctionDefinition def,
-			StdumpAST.SourceFile sourceFile, Address low) {
-		// Remove spam like "gcc2_compiled." and remove the existing label for
-		// the function name so it can be reapplied below.
-		SymbolTable symbolTable = program.getSymbolTable();
-		Symbol[] existingSymbols = symbolTable.getSymbols(low);
-		for(Symbol existingSymbol : existingSymbols) {
-			String name = existingSymbol.getName();
-			if(name.equals("__gnu_compiled_cplusplus") || name.equals("gcc2_compiled.") || name.equals(def.name)) {
-				symbolTable.removeSymbolSpecial(existingSymbol);
-			}
-		}
-		
-		// Ghidra will sometimes find the wrong label and use it as a function
-		// name e.g. "gcc2_compiled." so it's important that we set the name
-		// explicitly here.
+			StdumpAST.SourceFile sourceFile, Address low, Program program) {
 		try {
 			function.setName(def.name, SourceType.ANALYSIS);
 		} catch (DuplicateNameException | InvalidInputException e) {
 			log.appendException(e);
 		}
-		function.setComment(sourceFile.path);
 	}
 	
 	private HashSet<String> fillInParameters(Function function, StdumpAST.ImporterState importer,

--- a/src/main/java/ghidra/emotionengine/symboltable/StdumpAST.java
+++ b/src/main/java/ghidra/emotionengine/symboltable/StdumpAST.java
@@ -277,7 +277,12 @@ public class StdumpAST {
 						if(node.name != null && node.name.equals("__vtable")) {
 							field = new PointerDataType(createVtable(importer));
 						} else {
-							node.prefix += name + "__";
+							if(prefix != null) {
+								node.prefix += prefix;
+							}
+							if(name != null) {
+								node.prefix += name + "__";
+							}
 							field = replaceVoidWithUndefined1(node.createType(importer));
 						}
 						addField(type, field, node, node.relativeOffsetBytes, importer);
@@ -287,6 +292,12 @@ public class StdumpAST {
 				Union type = (Union) dest;
 				for(Node node : fields) {
 					if(node.storageClass != StorageClass.STATIC) {
+						if(prefix != null) {
+							node.prefix += prefix;
+						}
+						if(name != null) {
+							node.prefix += name + "__";
+						}
 						DataType field = replaceVoidWithUndefined1(node.createType(importer));
 						type.add(field, field.getLength(), node.name, "");
 					}

